src/main/java/com/programmers/springbootbasic/domain/customer/vo/Email.java

```diff
@@ -0,0 +1,23 @@
+package com.programmers.springbootbasic.domain.customer.vo;
+
+import com.programmers.springbootbasic.domain.customer.exception.ErrorMsg;
+import lombok.Getter;
+
+import java.util.regex.Pattern;
+
+@Getter
+public class Email {
+    private static final String EMAIL_REGEX = "^[a-zA-Z0-9_+&*-]+(?:\\.[a-zA-Z0-9_+&*-]+)*@(?:[a-zA-Z0-9-]+\\.)+[a-zA-Z]{2,7}$";
```

Comment:
```suggestion
    private static final String EMAIL_REGEX = "^[a-zA-Z0-9_+&*-]+(?:\\.[a-zA-Z0-9_+&*-]+)*@(?:[a-zA-Z0-9-]+\\.)+[a-zA-Z]{2,7}$";
    private static final Pattern EMAIL_PATTERN = Pattern.compile(EMAIL_REGEX);
```

Pattern 클래스의 compile() 메서드가 생각보다 높은 비용이 드는 작업이라고 합니다.
따라서 Pattern.compile()메서드를 상수로 선언하고 재활용하여 비용을 줄이는 방법도 좋을 것 같습니다.
아래 키워드로 검색해보시면 참고할만한 많은 블로그가 나올 것 같습니다.!

- 이펙티브 자바 아이템 6 불필요한 객체 생성을 피하라